### ISSUE TITLE
Plugin does not support vite@5.1.0 or higher versions

### ISSUE THREAD

**Issue by fdc-viktor-luft** [Dschungelabenteuer/vite-plugin-entry-shaking#39]
**Describe the bug** The plugin worked until version 5.0.12 of vite. Starting from version 5.1.0 it is not allowing to start the app in development mode anymore.

**To Reproduce** Try running the Vite dev mode with the plugin and vite@5.1.0

**Expected behavior** The plugin continues to function as before.

**Additional context** Almost immediately after starting the dev mode the following kind of error appears in the terminal:

```
error when starting dev server:
Error: Could not analyze entry file "...<private>..."
    at file:///<private>/node_modules/.pnpm/vite-plugin-entry-shaking@0.3.3_vite@5.1.0/node_modules/vite-plugin-entry-shaking/dist/index.mjs:336:11
    at async Object.analyzeEntry (file:///<private>/node_modules/.pnpm/vite-plugin-entry-shaking@0.3.3_vite@5.1.0/node_modules/vite-plugin-entry-shaking/dist/index.mjs:335:3)
    at async file:///<private>/node_modules/.pnpm/vite-plugin-entry-shaking@0.3.3_vite@5.1.0/node_modules/vite-plugin-entry-shaking/dist/index.mjs:344:7
    at async Promise.all (index 0)
    at async Object.analyzeEntries (file:///<private>/node_modules/.pnpm/vite-plugin-entry-shaking@0.3.3_vite@5.1.0/node_modules/vite-plugin-entry-shaking/dist/index.mjs:341:3)
    at async configResolved (file:///<private>/node_modules/.pnpm/vite-plugin-entry-shaking@0.3.3_vite@5.1.0/node_modules/vite-plugin-entry-shaking/dist/index.mjs:370:17)
    at async Promise.all (index 2)
    at async resolveConfig (file:///<private>/node_modules/.pnpm/vite@5.1.0_@types+node@20.11.17_sass@1.70.0/node_modules/vite/dist/node/chunks/dep-nGG-_oRu.js:67848:5)
    at async _createServer (file:///<private>/node_modules/.pnpm/vite@5.1.0_@types+node@20.11.17_sass@1.70.0/node_modules/vite/dist/node/chunks/dep-nGG-_oRu.js:64223:20)
    at async CAC.<anonymous> (file:///<private>/node_modules/.pnpm/vite@5.1.0_@types+node@20.11.17_sass@1.70.0/node_modules/vite/dist/node/cli.js:762:24)
```

**Comment by Dschungelabenteuer** [comment]
Hey! Thank's again for your feedback, there's been some work done late 2023 (and last month) to address most of the issues that were opened (including yours!). Unfortunately, I could not find much time to get it 100% ready, Apologies for the delay, I'll need to get back to it soon. In the mean time, could you by any chance test your codebase against the [`next` branch](https://github.com/Dschungelabenteuer/vite-plugin-entry-shaking/tree/next)? I've tried to run it on the [simple example](https://github.com/Dschungelabenteuer/vite-plugin-entry-shaking/tree/next/examples/simple) and did not notice any error (but as the name states, it's a rather trivial example and I may be missing something).

_(If you were to test it by locally by cloning & building the next branch, please don't mind the "debugger app" that may spawn along with Vite's dev server as it is still WIP)_

**Comment by fdc-viktor-luft** [comment]
Will do and come back to you with the results. Thanks for the great news :+1:

**Comment by fdc-viktor-luft** [comment]
Can you release a "next" version using `npm publish --tag next`?

**Comment by Dschungelabenteuer** [comment]
> Can you release a "next" version using `npm publish --tag next`?

[Sure thing, here you go (before I get distracted yet again!)](https://www.npmjs.com/package/vite-plugin-entry-shaking/v/0.0.0-next-20240219224802)

I also restored the example dedicated to your issue :)

**Comment by fdc-viktor-luft** [comment]
Used versions:

```
    "vite": "5.1.3",
    "vite-plugin-entry-shaking": "0.0.0-next-20240219224802",
```

Sadly, starting locally still results in an error:

```
Error: EISDIR: illegal operation on a directory, read
    at readFileSync (node:fs:453:20)
    at file:///<private>/node_modules/.pnpm/vite-plugin-entry-shaking@0.0.0-next-20240219224802_vite@5.1.3_vue@3.4.19/node_modules/vite-plugin-entry-shaking/dist/index.mjs:210:14
    at async Context.measure (file:///<private>/node_modules/.pnpm/vite-plugin-entry-shaking@0.0.0-next-20240219224802_vite@5.1.3_vue@3.4.19/node_modules/vite-plugin-entry-shaking/dist/index.mjs:674:17)
    at async Object.doAnalyzeEntry (file:///<private>/node_modules/.pnpm/vite-plugin-entry-shaking@0.0.0-next-20240219224802_vite@5.1.3_vue@3.4.19/node_modules/vite-plugin-entry-shaking/dist/index.mjs:208:20)
    at async Object.analyzeEntry (file:///<private>/node_modules/.pnpm/vite-plugin-entry-shaking@0.0.0-next-20240219224802_vite@5.1.3_vue@3.4.19/node_modules/vite-plugin-entry-shaking/dist/index.mjs:195:10)
    at async file:///<private>/node_modules/.pnpm/vite-plugin-entry-shaking@0.0.0-next-20240219224802_vite@5.1.3_vue@3.4.19/node_modules/vite-plugin-entry-shaking/dist/index.mjs:183:7
    at async Promise.all (index 0)
    at async file:///<private>/node_modules/.pnpm/vite-plugin-entry-shaking@0.0.0-next-20240219224802_vite@5.1.3_vue@3.4.19/node_modules/vite-plugin-entry-shaking/dist/index.mjs:181:5
    at async Context.measure (file:///<private>/node_modules/.pnpm/vite-plugin-entry-shaking@0.0.0-next-20240219224802_vite@5.1.3_vue@3.4.19/node_modules/vite-plugin-entry-shaking/dist/index.mjs:674:17)
    at async Object.analyzeEntries (file:///<private>/node_modules/.pnpm/vite-plugin-entry-shaking@0.0.0-next-20240219224802_vite@5.1.3_vue@3.4.19/node_modules/vite-plugin-entry-shaking/dist/index.mjs:179:31) {
  errno: -21,
  code: 'EISDIR',
  syscall: 'read'
}
error when starting dev server:
Error: Could not analyze entry file "/<private>/ui/components/"
    at file:///<private>/node_modules/.pnpm/vite-plugin-entry-shaking@0.0.0-next-20240219224802_vite@5.1.3_vue@3.4.19/node_modules/vite-plugin-entry-shaking/dist/index.mjs:199:11
    at async Object.analyzeEntry (file:///<private>/node_modules/.pnpm/vite-plugin-entry-shaking@0.0.0-next-20240219224802_vite@5.1.3_vue@3.4.19/node_modules/vite-plugin-entry-shaking/dist/index.mjs:195:10)
    at async file:///<private>/node_modules/.pnpm/vite-plugin-entry-shaking@0.0.0-next-20240219224802_vite@5.1.3_vue@3.4.19/node_modules/vite-plugin-entry-shaking/dist/index.mjs:183:7
    at async Promise.all (index 0)
    at async file:///<private>/node_modules/.pnpm/vite-plugin-entry-shaking@0.0.0-next-20240219224802_vite@5.1.3_vue@3.4.19/node_modules/vite-plugin-entry-shaking/dist/index.mjs:181:5
    at async Context.measure (file:///<private>/node_modules/.pnpm/vite-plugin-entry-shaking@0.0.0-next-20240219224802_vite@5.1.3_vue@3.4.19/node_modules/vite-plugin-entry-shaking/dist/index.mjs:674:17)
    at async Object.analyzeEntries (file:///<private>/node_modules/.pnpm/vite-plugin-entry-shaking@0.0.0-next-20240219224802_vite@5.1.3_vue@3.4.19/node_modules/vite-plugin-entry-shaking/dist/index.mjs:179:31)
    at async Context.init (file:///<private>/node_modules/.pnpm/vite-plugin-entry-shaking@0.0.0-next-20240219224802_vite@5.1.3_vue@3.4.19/node_modules/vite-plugin-entry-shaking/dist/index.mjs:663:20)
    at async configResolved (file:///<private>/node_modules/.pnpm/vite-plugin-entry-shaking@0.0.0-next-20240219224802_vite@5.1.3_vue@3.4.19/node_modules/vite-plugin-entry-shaking/dist/index.mjs:726:7)
    at async Promise.all (index 2)
 ELIFECYCLE  Command failed with exit code 1.
```

**Comment by fdc-viktor-luft** [comment]
Yes, there is a `/ui/components/index.ts`.

`.tsconfig`

```json
{
  "extends": "../../.tsconfig.base.json",
  "compilerOptions": {
    "baseUrl": "."
  },
  "include": [
    "**/*.ts",
    "**/*.vue"
  ],
  "exclude": [
    "node_modules"
  ]
}
```

And `.tsconfig.base.json`

```json
{
  "compilerOptions": {
    "target": "es2020",
    "module": "esnext",
    "strict": true,
    "jsx": "preserve",
    "importHelpers": true,
    "moduleResolution": "node",
    "skipLibCheck": true,
    "esModuleInterop": true,
    "allowSyntheticDefaultImports": true,
    "sourceMap": true,
    "resolveJsonModule": true,
    "noEmit": true,
    "lib": [
      "esnext",
      "dom",
      "dom.iterable",
      "scripthost"
    ],
    "paths": {
      "@/*": [
        "./src/*"
      ]
    }
  }
}
```

Node Version: v20.11.0

**Comment by Dschungelabenteuer** [comment]
I'm definitely having a hard time reproducing it, even by using your whole merged tsconfig and using the same node version. I've tried leaving trailing slashes at the end of entry paths, using aliases, mixed module extensions, etc. Tried on Linux and Windows environments. 

Maybe there's something with your `getAllLibs` function? Is there any chance I could get a preview of its implementation? Anyway, I need to make a slight change to the fix I made for your #28 issue so I'll need to release another next tag for you to test it. If you have an idea of some kind of log that could help find the cause of your issue, please let me know!

**Comment by fdc-viktor-luft** [comment]
Ok, I just wanted to debug the root cause, but installed a later version of Vite now:

```json
    "vite": "5.1.4",
    "vite-plugin-entry-shaking": "0.0.0-next-20240219224802",
```

It works without any issues. It also feels way faster than the older versions 🤔 

So, if you also think, we should not waste time investigating why it didn't work with the previous Vite version, I would be happy to just expect your next release :)

**Comment by fdc-viktor-luft** [comment]
Do you already know roughly when the new release will be available?

**Comment by Dschungelabenteuer** [comment]
Next version's core has been ready for quite some time now but I initially really wanted to get the upcoming debugger on point before I release anything. But tbh, I won't ever be satisfied enough with it so II'll probably take two weeks to do some fine-tuning and eventually release it even if debugger's not 100% complete. I'd say most probably ~March 30th.

**Comment by Dschungelabenteuer** [comment]
@fdc-viktor-luft I've just released v0.4.0, I'm closing the issue but feel free to re-open if needed :-)
